perceval/converters/myqlm_converter.py

```diff
@@ -0,0 +1,151 @@
+from perceval.utils import P, BasicState, Encoding
+from perceval.utils.algorithms.optimize import optimize
+from perceval.utils.algorithms.norm import frobenius
+import perceval.components.unitary_components as comp
+import numpy as np
+
+
+min_precision_gate = 1e-4
+
+
+class MyQLMConverter:
+    r"""myQLM quantum circuit to perceval circuit converter.
+    :param catalog: component library of perceval
+    """
+    def __init__(self, catalog, backend_name: str = "SLOS", source: Source = Source()):
+        self._source = source
+        self._backend_name = backend_name
+        self._heralded_cnot_builder = catalog["heralded cnot"]
+        self._heralded_cz_builder = catalog["heralded cz"]
+        self._postprocessed_cnot_builder = catalog["postprocessed cnot"]
+        self._generic_2mode_builder = catalog["generic 2 mode circuit"]
+        self._lower_phase_component = Circuit(2) // (0, comp.PS(P("phi2")))
+        self._upper_phase_component = Circuit(2) // (1, comp.PS(P("phi1")))
+        self._two_phase_component = Circuit(2) // (0, comp.PS(P("phi1"))) // (1, comp.PS(P("phi2")))
+
+    def convert(self, qlmc, use_postselection: bool = True) -> Processor:
+        r"""Convert a myQLM quantum circuit into a perceval `Processor`.
+
+        :param qlmc: quantum gate-based myqlm circuit
+        :type qlmc: qat.core.Circuit
+        :param use_postselection: when True, uses a `post-processed CNOT` as the last gate. Otherwise, uses only
+            `heralded CNOT`
+        :return: the converted Processor
+        """
+        import qat
+        from qat.core.circuit_builder.matrix_util import circ_to_np
+        # importing the quantum toolbox of myqlm
+        # this nested import fixes automatic class reference generation
+
+        # count the number of CNOT gates to use during the conversion, will give us the number of herald to handle
+        n_cnot = qlmc.count("CNOT")
+        cnot_idx = 0
+
+        n_moi = qlmc.nbqbits * 2  # number of modes of interest = 2 * number of qbits
+        input_list = [0] * n_moi
+        p = Processor(self._backend_name, n_moi, self._source)
+
+        for i in range(qlmc.nbqbits):
+            p.add_port(i * 2, Port(Encoding.DUAL_RAIL, f'Q{i}'))
+            input_list[i * 2] = 1
+        default_input_state = BasicState(input_list)
+
+        for i, instruction in enumerate(qlmc.iterate_simple()):
+            instruction_name = instruction[0]  # name of the Gate
+            instruction_qbit = instruction[-1]  # tuple with list of qbit positions
+            # information carried by instruction
+            # tuple ('Name', [value of the parameter for gate], [list of qbit positions where gate is applied])
+
+            assert instruction_name in qlmc.gate_set, "cannot convert (%s)" % instruction_name
+            # only gates are converted -> checking if instruction is in gate_set of AQASM
+
+            if len(instruction_qbit) == 1:
+                if instruction_name == "H":
```

Comment:
You should declare `ins` (`ins=None`) before the if statement line 93.
First some language work like that (see RAII) and it's a good idea to start coding like that now.
Then if you use this variable elsewhere in your code in the futur you can (and usually will) have unwanted  side effect because your variable will already be assign.
On reading your code, a fellow developper can immediatly spot that this condition statement is about building the right `ins`.